src/main/java/org/terasology/scenario/internal/systems/RegionTreeSystem.java

```diff
@@ -87,16 +101,17 @@ public void onRegionTreeDeleteEvent(RegionTreeDeleteEvent event, EntityRef entit
 
         event.getDeleteEntity().destroy();
 
-        if (event.getHubScreen() != null) {
-            event.getHubScreen().updateRegionTree(entity);
+        for (EntityRef e : entityManager.getEntitiesWith(ScenarioHubToolUpdateComponent.class)) {
+            e.getComponent(ScenarioHubToolUpdateComponent.class).dirtyRegions = true;
+            e.saveComponent(e.getComponent(ScenarioHubToolUpdateComponent.class));

```

Comment:
I think e.getComponent(ScenarioHubToolUpdateComponent.class) should be extracted to a local variable. This does not just save a lookup and makes the code more readable but also prevents future bugs when for example it would get changed to getComponent returning a fresh copy instead of the same instance.